src/prefect/orion/orchestration/rules.py

```diff
@@ -384,6 +392,12 @@ async def _validate_proposed_state(
         db: OrionDBInterface,
     ):
         if self.proposed_state is None:
+            # If the run has no state and needs to stay in the same state, there is
+            # nothing to do here
+            if self.run.state is None:
+                self.validated_state = None
+                return
```

Comment:
This was previous behavior but avoids calling `self.session.add(None)` which throws an exception